openghg/store/metastore/_metastore.py

```diff
@@ -0,0 +1,17 @@
+from __future__ import annotations
+from openghg.store.base import BaseStore, get_data_class
+
+
+def open_metastore(bucket: str, data_type: str) -> BaseStore:
```

Comment:
Yea it's just meant as a way to isolate when `BaseStore` is being used to access the metastore, and in the next PR, I swap over how that works to a `TinyDBMetaStore` class.

I'm still making a few changes, but you can see that here: https://github.com/openghg/openghg/pull/771/files#diff-5e46d0ea9f0221bd9341d6c61b7fb396cbcc3d31fceb69f4056d03d4002c5294. There are some loose ends still because some parts of the tests and _integrity module use a `datasources()` method or access `_datasource_uuids`, which depend on some parts of `BaseStore` that aren't in the metastore. (Gareth has removed one of these in the UUIDs PR, where datasource UUIDs used to be stored with file hashes.)

The 4th PR will be an `ObjectStore` class that takes children of a `MetaStore` ABC and a `Datasource` ABC as parameters. At that point, it might be worth just replacing `open_metastore` with `load_store` or `open_store`, and the parts that rely on `datasources()` or `_datasource_uuids` can be fixed up at that point. 

Then that leaves `BaseStore` children to: check the data schema, check the metadata/datasource schema (required keys), and call parser functions to make data/metadata adhere to the schemas. (And also track file hashes, but that's kind of a minor thing.)

I can drop this PR and just ask for part 3 to be reviewed in a day or two.